irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -247,6 +248,31 @@ namespace iroha {
           });
     }
 
+    rxcpp::observable<model::Transaction> StorageImpl::getAccountAssetTransactions(
+        std::string account_id, std::string asset_id) {
+      std::shared_lock<std::shared_timed_mutex> write(rw_lock_);
+      return getBlocksFrom(1)
+          .flat_map([](auto block) {
+            return rxcpp::observable<>::iterate(block.transactions);
+          })
+          .map([account_id, asset_id](model::Transaction tx) {
+            std::vector<std::shared_ptr<model::Command>> assetCommands;
+            std::for_each(tx.commands.begin(), tx.commands.end(), [account_id, asset_id, &assetCommands](std::shared_ptr<model::Command> command) {
+              if (instanceof <iroha::model::TransferAsset>(*command)) {
+                auto transferAsset = (iroha::model::TransferAsset*) command.get();
+                if ((transferAsset->src_account_id == account_id || transferAsset->dest_account_id == account_id) && transferAsset->asset_id == asset_id) {
+                  assetCommands.push_back(command);
+                }
+              }
+            });
+            tx.commands = assetCommands;
```

Comment:
It is not correct to modify transactions, since they are signed and immutable. One way I see is to return all transactions which contain at least one transfer. Another way is to return observable of commands.
Also it is better to reuse getAccountTransactions with addition of `.filter()`.

Reply:
@lebdron Thank you! I'll return transactions that contain at lease one TransferAsset command with out editing data in the transaction.

And I'll use `.filter()`. Thanks!

Reply:
@cimadai please, also check tests and try to get rid of the code, which deletes signatories in this pull request. Thank you for your contribution to the project

Reply:
@neewy OK, I'll re-create new branch and new PR.